aggregate_enclave_scripts/controller.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/python3
+import sys
+import os
+import requests
+from flask import Flask, request, Response, jsonify
+import json
+import uuid
+import threading
+import subprocess
+import docker
+from docker.types import Mount
+json_data = json.load(open("mock_data.json"))
+list_of_containers = list(json.load(open("mock_data.json")).keys())
+client = docker.from_env()
+
+app = Flask(__name__)
+path = os.path.expanduser("~/e-mission-server/")
+# container_port = 1025
+
+class DockerThread(threading.Thread):
+        def __init__(self, container, query_type, uuid, agg_ip, privacy_budget):
+                threading.Thread.__init__(self)
+                self.container = container
+                self.query_type = query_type
+                self.uuid = uuid
+                self.agg_ip = agg_ip
+                self.privacy_budget = privacy_budget
+
+        def run(self):
+                self.container.unpause()
+                output = self.container.exec_run('bash bash_file ' + self.query_type + ' ' + self.uuid + ' ' + self.agg_ip + ' ' + self.privacy_budget)
+                print(output)
+                self.container.pause()
+
+@app.route('/', methods=['GET'])
+def home():
+        return "hello!"
+
+@app.route('/upload_info', methods=['POST'])
+def upload():
+        pass
+
+@app.route('/remove_containers', methods=['GET'])
+def remove_containers():
+    for container in list_of_containers:
+        container[0].remove(force=True)
+    return ""
+@app.route('/request_query', methods=['POST'])
+def query_start():
+        """
+        1. Read list of enclaves from file
+        2. Wake them up with docker resume
+        3. Ask for query from them
+        """
+        request_dict = json.loads(request.data.decode('utf-8'))
+        query_type = str(request_dict['query_type'])
+        privacy_budget = str(request_dict['privacy_budget'])
+        print(request_dict)
+        threads = []
+        aggregator_ip = request.environ['REMOTE_ADDR'] + ':2001'
+        print("aggregator_ip: " + str(aggregator_ip))
+        print("Length of list of containers: " + str(len(list_of_containers)))
+        batch_size = 10
```

Comment:
Yeah so I think most recently we tried going up to 20 just to test it again, and at some point it crashed the entire computer :/

Reply:
Could this be because the SGX memory is limited?
https://software.intel.com/en-us/forums/intel-software-guard-extensions-intel-sgx/topic/670322

But according to that thread, you can get upto 4GB of enclave memory with the Linux driver. Is there a similar limit in simulation mode?

Reply:
Now that we are using non-enclave docker images, should be able to experiment with much larger numbers